liblouis/compileTranslationTable.c

```diff
@@ -2776,51 +2773,50 @@ compileRule(FileInfo *nested, TranslationTableHeader **table,
 			getCharacters(nested, &ptn_after);
 			getRuleDotsPattern(nested, &ruleDots);
 
-			if (!addRule(nested, opcode, &ruleChars, &ruleDots, after, before,
-						&ruleOffset, &rule, noback, nofor, table)) {
-				ok = 0;
-				break;
-			}
-			if (ptn_before.chars[0] == '-' && ptn_before.length == 1)
-				len = _lou_pattern_compile(
-						&ptn_before.chars[0], 0, &patterns[1], 13841, *table, nested);
-			else
-				len = _lou_pattern_compile(&ptn_before.chars[0], ptn_before.length,
-						&patterns[1], 13841, *table, nested);
-			if (!len) {
-				ok = 0;
-				break;
-			}
-			mrk = patterns[0] = len + 1;
-			_lou_pattern_reverse(&patterns[1]);
-
-			if (ptn_after.chars[0] == '-' && ptn_after.length == 1)
-				len = _lou_pattern_compile(
-						&ptn_after.chars[0], 0, &patterns[mrk], 13841, *table, nested);
-			else
-				len = _lou_pattern_compile(&ptn_after.chars[0], ptn_after.length,
-						&patterns[mrk], 13841, *table, nested);
-			if (!len) {
-				ok = 0;
-				break;
-			}
-			len += mrk;
-
-			if (!allocateSpaceInTranslationTable(
-						nested, &patternsOffset, len * sizeof(widechar), table)) {
-				ok = 0;
-				break;
+			if (addRule(nested, opcode, &ruleChars, &ruleDots, after, before, &ruleOffset,
```

Comment:
Personally I like the early return style better (as you have it in other places). For me it makes the code more readable. More so than deeply nested if statements. For example:
``` C
if (!addRule(nested, opcode, &ruleChars, &ruleDots, after, before, &ruleOffset, &rule, noback, nofor, table))
   return 0;
// happy path
if (!len) return 0;
// more happy path
```

Reply:
I may have done this because I wanted to perform a cleanup step, notably the `free(patterns)` at the end. I guess I could have used a `goto` instead.

I agree that the early return style is sometimes more readable, but jumps can also make code less readable. You have to be careful with them. In this case both styles are probably fine.

Reply:
Ah, I see. Hm, tough call. I seem to remember that `goto cleanup` or some such is used in other places, so it might be ok.

Reply:
I'm not sure what to do now. Different styles are used in different places, so whatever I do to the code that I touched in this commit, it will still seem random when you look at the whole code base. Or do you only have a problem with nested `if`s if it becomes too deep?

Reply:
Only if they're too deep.

Reply:
Hi @egli, I decided to go through the whole file and use the early return style everywhere. I squashed it with the other commit in https://github.com/liblouis/liblouis/pull/1088/commits/6f2432be47b15f716d99f88b2e4f657b79a3e295.